### PR TITLE
Temporarily pin glue-core<1.19.0 to fix test failure

### LINKED ISSUE #2805: TST: test_cubeviz_aperphot_cube_orig_flux fails with RuntimeError: NULL error object in wcslib
- No need to revert https://github.com/spacetelescope/jdaviz/pull/2809

Example log: https://github.com/spacetelescope/jdaviz/actions/runs/8706207800/job/23883214225

I started noticing this after merging #2802 but that PR had all green CI checks. I can also reproduce this error locally though it was initially green but only started failing after I reinstalled some packages, so the green was probably due to an outdated cache somewhere. This test does not use any remote data. It is not immediately clear to me why it is suddenly failing and why the error is all the way in `wcslib`.

The test:

https://github.com/spacetelescope/jdaviz/blob/620ccfed55b1209b737377a272111c343eb2b2e5/jdaviz/configs/cubeviz/plugins/tests/test_cubeviz_aperphot.py#L10

The offending call:

https://github.com/spacetelescope/jdaviz/blob/620ccfed55b1209b737377a272111c343eb2b2e5/jdaviz/configs/cubeviz/plugins/tests/test_cubeviz_aperphot.py#L62-L63

The data used:

https://github.com/spacetelescope/jdaviz/blob/620ccfed55b1209b737377a272111c343eb2b2e5/jdaviz/conftest.py#L102-L135

```
_____________________ test_cubeviz_aperphot_cube_orig_flux _____________________

cubeviz_helper = <jdaviz.configs.cubeviz.helper.Cubeviz object at 0x7fcef5e5cfb0>
image_cube_hdu_obj_microns = [<astropy.io.fits.hdu.image.PrimaryHDU object at 0x7fceeb071820>, <astropy.io.fits.hdu.image.ImageHDU object at 0x7fce...py.io.fits.hdu.image.ImageHDU object at 0x7fceeb073530>, <astropy.io.fits.hdu.image.ImageHDU object at 0x7fceeb073b60>]

    def test_cubeviz_aperphot_cube_orig_flux(cubeviz_helper, image_cube_hdu_obj_microns):
        cubeviz_helper.load_data(image_cube_hdu_obj_microns, data_label="test")
        flux_unit = u.Unit("1E-17 erg*s^-1*cm^-2*Angstrom^-1")
    
        aper = RectanglePixelRegion(center=PixCoord(x=1, y=2), width=3, height=5)
        cubeviz_helper.load_regions(aper)
    
        # Make sure MASK is not an option even when shown in viewer.
        cubeviz_helper.app.add_data_to_viewer("flux-viewer", "test[MASK]", visible=True)
    
        plg = cubeviz_helper.plugins["Aperture Photometry"]._obj
        assert plg.dataset.labels == ["test[FLUX]", "test[ERR]"]
        assert plg.cube_slice == "4.894e+00 um"
    
        plg.dataset_selected = "test[FLUX]"
        plg.aperture_selected = "Subset 1"
        plg.vue_do_aper_phot()
        row = cubeviz_helper.get_aperture_photometry_results()[0]
    
        # Basically, we should recover the input rectangle here.
        assert_allclose(row["xcenter"], 1 * u.pix)
        assert_allclose(row["ycenter"], 2 * u.pix)
        sky = row["sky_center"]
        assert_allclose(sky.ra.deg, 205.43985906934287)
        assert_allclose(sky.dec.deg, 27.003490103642033)
        assert_allclose(row["sum"], 75 * flux_unit)  # 3 (w) x 5 (h) x 5 (v)
        assert_allclose(row["sum_aper_area"], 15 * (u.pix * u.pix))  # 3 (w) x 5 (h)
        assert_allclose(row["mean"], 5 * flux_unit)
        assert_quantity_allclose(row["slice_wave"], 4.894499866699333 * u.um)
    
        # Move slider and make sure it recomputes for a new slice automatically.
        cube_slice_plg = cubeviz_helper.plugins["Slice"]._obj
        cube_slice_plg.vue_goto_first()
        plg.vue_do_aper_phot()
        row = cubeviz_helper.get_aperture_photometry_results()[1]
    
        # Same rectangle but different slice value.
        assert_allclose(row["xcenter"], 1 * u.pix)
        assert_allclose(row["ycenter"], 2 * u.pix)
        sky = row["sky_center"]
        assert_allclose(sky.ra.deg, 205.43985906934287)
        assert_allclose(sky.dec.deg, 27.003490103642033)
        assert_allclose(row["sum"], 15 * flux_unit)  # 3 (w) x 5 (h) x 1 (v)
        assert_allclose(row["sum_aper_area"], 15 * (u.pix * u.pix))  # 3 (w) x 5 (h)
        assert_allclose(row["mean"], 1 * flux_unit)
        assert_quantity_allclose(row["slice_wave"], 4.8904998665093435 * u.um)
    
        # We continue on with test_cubeviz_aperphot_generated_2d_collapse here
        # because we want to make sure the result would append properly between 3D and 2D.
        collapse_plg = cubeviz_helper.plugins["Collapse"]._obj
        collapse_plg.vue_collapse()
    
        # Need this to make it available for photometry data drop-down.
>       cubeviz_helper.app.add_data_to_viewer("uncert-viewer", "test[FLUX] collapsed")

../../.tox/py312-test-devdeps/lib/python3.12/site-packages/jdaviz/configs/cubeviz/plugins/tests/test_cubeviz_aperphot.py:63: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/jdaviz/app.py:1498: in add_data_to_viewer
    self.set_data_visibility(viewer_item, data_label, visible=visible, replace=clear_other_data)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/jdaviz/app.py:2032: in set_data_visibility
    viewer.add_data(data, percentile=95, color=viewer.color_cycler())
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue_jupyter/view.py:118: in add_data
    result = super().add_data(data)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/common/viewer.py:209: in add_data
    layer = get_layer_artist_from_registry(data, self) or self.get_data_layer_artist(data)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue_jupyter/bqplot/image/viewer.py:101: in get_data_layer_artist
    return self.get_layer_artist(cls, layer=layer, layer_state=layer_state)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue_jupyter/view.py:143: in get_layer_artist
    return cls(self, self.state, layer=layer, layer_state=layer_state)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue_jupyter/bqplot/image/layer_artist.py:21: in __init__
    super().__init__(view, *args, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/image/layer_artist.py:68: in __init__
    super(ImageLayerArtist, self).__init__(axes, viewer_state,
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/image/layer_artist.py:31: in __init__
    super(BaseImageLayerArtist, self).__init__(axes, viewer_state,
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/matplotlib/layer_artist.py:18: in __init__
    super(MatplotlibLayerArtist, self).__init__(viewer_state,
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/common/layer_artist.py:25: in __init__
    self.state = layer_state or self._layer_state_cls(viewer_state=viewer_state,
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue_jupyter/bqplot/image/state.py:34: in __init__
    super(BqplotImageLayerState, self).__init__(*args, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/image/state.py:550: in __init__
    self._update_attribute()
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/image/state.py:559: in _update_attribute
    self.attribute_att_helper.set_multiple_data([self.layer])
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/core/data_combo_helper.py:322: in set_multiple_data
    self.refresh()
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/core/data_combo_helper.py:379: in refresh
    self.choices = choices
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/core/data_combo_helper.py:84: in choices
    with delay_callback(self.state, self.selection_property):
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/echo/core.py:538: in __exit__
    p.notify(*args)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/utils/matplotlib.py:170: in wrapper
    return func(*args, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/matplotlib/state.py:38: in notify
    super(DeferredDrawSelectionCallbackProperty, self).notify(*args, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/echo/core.py:125: in notify
    cback(new)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/image/state.py:608: in _update_attribute_display_unit_choices
    self.attribute_display_unit = component.units
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/echo/core.py:263: in __setattr__
    self._notify_global(**{attribute: value})
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/utils/matplotlib.py:170: in wrapper
    return func(*args, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/viewers/matplotlib/state.py:319: in _notify_global
    super(MatplotlibLayerState, self)._notify_global(*args, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/echo/core.py:258: in _notify_global
    callback(**kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/core/state_objects.py:206: in _update_values
    self.update_values(**properties)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/core/state_objects.py:349: in update_values
    lower, upper = converter.to_unit(self.data,
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/core/units.py:31: in to_unit
    return self.converter_helper.to_unit(data, cid, values, original_units, target_units)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/jdaviz/app.py:98: in to_unit
    spec = data.get_object(cls=Spectrum1D)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue/core/data.py:298: in get_object
    return handler.to_object(self, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/glue_astronomy/translators/spectrum1d.py:287: in to_object
    return Spectrum1D(**data_kwargs, **kwargs)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/specutils/spectra/spectrum1d.py:277: in __init__
    if hasattr(self.wcs, "spectral"):
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/astropy/wcs/wcs.py:3488: in spectral
    return self.sub([WCSSUB_SPECTRAL])  # Defined by C-ext
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/astropy/wcs/wcs.py:656: in sub
    copy = self.deepcopy()
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/astropy/wcs/wcs.py:653: in deepcopy
    return copy.deepcopy(self)
/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/copy.py:143: in deepcopy
    y = copier(memo)
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/astropy/wcs/wcs.py:626: in __deepcopy__
    deepcopy(self.wcs, memo),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

x =        flag: 137
      naxis: 0
      crpix: 0x5632efcd0720
            
         pc: 0x5632efcbf970
      cdelt: 0x56...(remainder unused)
    isGrism: 0
        err: 0x0
     spxX2P: 0x0
     spxP2S: 0x0
     spxS2P: 0x0
     spxP2X: 0x0

memo = {}, _nil = []

    def deepcopy(x, memo=None, _nil=[]):
        """Deep copy operation on arbitrary Python objects.
    
        See the module's __doc__ string for more info.
        """
    
        if memo is None:
            memo = {}
    
        d = id(x)
        y = memo.get(d, _nil)
        if y is not _nil:
            return y
    
        cls = type(x)
    
        copier = _deepcopy_dispatch.get(cls)
        if copier is not None:
            y = copier(x, memo)
        else:
            if issubclass(cls, type):
                y = _deepcopy_atomic(x, memo)
            else:
                copier = getattr(x, "__deepcopy__", None)
                if copier is not None:
>                   y = copier(memo)
E                   RuntimeError: NULL error object in wcslib

/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/copy.py:143: RuntimeError
=========================== short test summary info ============================
FAILED ../../.tox/py312-test-devdeps/lib/python3.12/site-packages/jdaviz/configs/cubeviz/plugins/tests/test_cubeviz_aperphot.py::test_cubeviz_aperphot_cube_orig_flux - RuntimeError: NULL error object in wcslib
```

[🐱](https://jira.stsci.edu/browse/JDAT-4410)

### PR DESCRIPTION
**Pull request by rosteen**
Fixes #2805 for now until we figure out why glue-core broke the test.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ dependencies = [
     "traitlets>=5.0.5",
     "bqplot>=0.12.37",
     "bqplot-image-gl>=1.4.11",
-    "glue-core>=1.17.1",
+    "glue-core>=1.17.1,<1.19.0",
```
Comment (pllim) [inline pyproject.toml:15]: If we do this, maybe we don't have to unpin it when/if Tom releases a 1.19.1 with a fix?
```suggestion
    "glue-core>=1.17.1,!=1.19.0",
```

Comment (rosteen) [reply pyproject.toml]: Good point. Unless 1.19.1 is also broken 😅

**Review by pllim — APPROVED** [review]
As long as CI is green (including devdeps). Thanks!

**Comment by rosteen** [comment]
> Hmm I guess devdeps failing makes sense since it isn't fixed yet.

Yeah, I was about to comment asking if we really wanted to pin an old commit in devdeps or something rather than see it failing (and pass once it's fixed upstream).

**Comment by pllim** [comment]
I think letting devdeps fail is ok for now. It doesn't block anything and reminds us that upstream is broken. Thanks!

**Pull request merged**